packages/axios-extension/src/abap/ui5-abap-repository-service.ts

```diff
@@ -0,0 +1,281 @@
+import { AxiosResponse, AxiosRequestConfig, AxiosError } from 'axios';
+import { readFileSync } from 'fs';
+import { prettyPrintError, prettyPrintMessage } from './message';
+import { ODataService } from '../base/odata-service';
+
+/**
+ * Application information object returned by the UI5 Repository service
+ */
+export interface AppInfo {
+    Name: string;
+    Package: string;
+    Description?: string;
+    Info?: string;
+    ZipArchive?: string;
+}
+
+/**
+ * Required configuration to deploy an application using the UI5 Repository service.
+ */
+export interface ApplicationConfig {
+    name: string;
+    description?: string;
+    package?: string;
+    transport?: string;
+}
+
+export const abapUrlReplaceMap = new Map([
+    [/\.abap\./, '.abap-web.'],
+    [/-api\.s4hana\.ondemand\.com/, '.s4hana.ondemand.com'],
+    [/-api\.saps4hanacloud\.cn/, '.saps4hanacloud.cn']
```

Comment:
The first replacement may be problematic, `.abap.` in a URL is not uncommon. We may inadvertently do the replacement.

For the rest, why do we need a allow-list? That does not look complete. John Long may know? I don't know his GH username 😬 
@tobiasqueck, @devinea or @IainSAP, would you guys know about this list or about John's username to tag him?

Reply:
@longieirl